src/hotspot/cpu/aarch64/globals_aarch64.hpp

```diff
@@ -110,7 +110,9 @@ define_pd_global(intx, InlineSmallCode,          1000);
   product(int, SoftwarePrefetchHintDistance, -1,                        \
           "Use prfm hint with specified distance in compiled code."     \
           "Value -1 means off.")                                        \
-          range(-1, 4096)
+          range(-1, 4096)                                               \
+  product(bool, UseROPProtection, false,                                \
+          "Use ROP based branch protection")                            \
```

Comment:
The description is not correct. It's protection against certain ROP-based attack techniques.

Reply:
I don't agree that this is incorrect, at least not for the stated reason. The flag switches on a protection mechanism that guards against ROP attacks. To my reading that does not imply it guards against all such attacks, merely that this is the nature of the protection it offers.

The description might still be considered incorrect for an unrelated reason. Its use of the adjectival phrase ROP based constitutes a transferred epithet, conflating the symptom with the medicine. In other words, the protection offered is not ROP based i.e. does not rely on an ROP technique. What it does is protect against ROP attacks. So, I'd suggest rewording to

    "Enable protection of branches against ROP attacks".

Florian, if you want to argue for rewording that to "Enable protection of branches against some categories of ROP attacks" or some other equivalently qualified variant please feel free to make a case. However, I don't think see any need to add that rider, nor any precedent in any of the other short descriptions provided in globals.hpp.

Reply:
I did mean the description, not the flag name.

Reply:
Yes, understood. I too was talking about the description even though I introduced my comment by talking about what the flag does.

Reply:
`"Protect branches against ROP attacks".`